worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -424,6 +421,24 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
     expiredShuffleKeys.asScala.foreach { shuffleKey =>
       logInfo(s"Cleanup expired shuffle $shuffleKey.")
       if (fileInfos.containsKey(shuffleKey)) {
+        fileInfos.get(shuffleKey).asScala.foreach {
+          case (_, fileInfo) => {
+            val dir = fileInfo.getFile.getParentFile.getParentFile.getParentFile
+            logDebug(s"Cleanup expired shuffle before ${workingDirWriters.get(dir).size()}")
+            if (workingDirWriters.get(dir) != null) {
+              val fileWriter = workingDirWriters.get(dir).get(fileInfo.getFile.getAbsolutePath)
+              if (fileWriter != null) {
+                fileWriter.destroy(new IOException(
+                  s"Destroy FileWriter ${fileWriter} caused by shuffle ${shuffleKey} expired."))
+                logDebug(s"Cleanup expired filelength:${fileWriter.getFileInfo.getFileLength}")
+              }
+            }
+            if (workingDirWriters.get(dir).size() == 0) {
+              workingDirWriters.remove(dir)
```

Comment:
there will be a concurrent problem if we put a new filewriter while we remove this dir.

Reply:
yes. it my be null. now fixed